sui-execution/latest/sui-move-natives/src/crypto/sui_state_proof.rs

```diff
@@ -278,12 +255,17 @@ pub fn sui_state_proof_verify_committee(
             Ok(address) => address,
             Err(_) => return Ok(NativeResult::err(cost, INVALID_TX)),
         };
+
+
+        // Ensure that the cap id is the same for all messages.
+        if cap_id_final != SuiAddress::ZERO && cap_id_final != cap_id {
+            return Ok(NativeResult::err(cost, INVALID_TX));
+        }
```

Comment:
This fixes the vulnerability, however, I wonder if this is the best design; an alternative option would be for the native function to get a list of messages and a cap ID, and to check that all messages are approved by that cap ID. This would allow, for example, a single transaction in Sui to approve messages for different dWallets. @alexanderludwig